src/any_agent/tools/mcp/frameworks/__init__.py

```diff
@@ -22,7 +22,19 @@
 
 
 def _get_mcp_server(mcp_tool: MCPParams, agent_framework: AgentFramework) -> MCPServer:
-    return TypeAdapter[MCPServer](MCPServer).validate_python(
+    mapping = {
+        AgentFramework.GOOGLE: GoogleMCPServer,
+        AgentFramework.LANGCHAIN: LangchainMCPServer,
+        AgentFramework.LLAMA_INDEX: LlamaIndexMCPServer,
+        AgentFramework.OPENAI: OpenAIMCPServer,
+        AgentFramework.AGNO: AgnoMCPServer,
+        AgentFramework.SMOLAGENTS: SmolagentsMCPServer,
+        AgentFramework.TINYAGENT: TinyAgentMCPServer,
+    }
+    cls = mapping.get(agent_framework)
+    if cls is None:
+        raise ValueError(f"Unsupported agent framework: {agent_framework}")
+    return TypeAdapter[cls](cls).validate_python(
```

Comment:
@njbrake what do you think of dropping the `TypeAdapter` and doing just `if agent_frameworks is X`  + `assert_never` as we do in similar places?